server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -197,12 +207,25 @@ public long getLength() {
 
         public static UploadedSegmentMetadata fromString(String uploadedFilename) {
             String[] values = uploadedFilename.split(SEPARATOR);
-            return new UploadedSegmentMetadata(values[0], values[1], values[2], Long.parseLong(values[3]));
+            UploadedSegmentMetadata metadata = new UploadedSegmentMetadata(values[0], values[1], values[2], Long.parseLong(values[3]));
+            if (values.length == 5) {
```

Comment:
Setting aside compatibility concerns with previous versions of the remote store code (which is experimental so we don't have to maintain compatibility), are there are any cases where an uploaded segment metadata would not have a corresponding "written by" version? If not, I would recommend to structure this code so that it is impossible to parse a string without the version.

Reply:
Good point. I will update this because I can't think of any case where we won't have the Lucene version uploaded in the metadata now with the new changes.
Thanks!

Reply:
Hi @andrross,
The new commits address this. Marking this conversation as resolved. Feel free to reopen if there's anything else needed here.
Thanks!